lib/IDE/ExprContextAnalysis.cpp

```diff
@@ -393,12 +393,16 @@ static void collectPossibleCalleesByQualifiedLookup(
       tyExpr->setType(nullptr);
   }
 
-  auto baseTyOpt = getTypeOfCompletionContextExpr(
-      DC.getASTContext(), &DC, CompletionTypeCheckKind::Normal, baseExpr, ref);
-  if (!baseTyOpt)
-    return;
-
-  auto baseTy = (*baseTyOpt)->getWithoutSpecifierType();
+  Type baseTy = baseExpr->getType();
+  if (!baseTy || baseTy->is<ErrorType>()) {
```

Comment:
Why is it safe to re-check when it's ErrorType?

Reply:
It *should* be safe for any expression as long as `SanitizeExpr` works correctly.
This change is kind of an optimization which avoid unnecessary re-typechecking.

Reply:
So this change does not affect the correctness of this patch?

Reply:
Right. The first patch fixes the crash. The second patch doesn't affect the result of it.